### ISSUE TITLE
BUG in Media manager when deleting files

### ISSUE THREAD

**Issue by plyusninva** [octobercms/october#2765]
##### October build 396

A bug occurred when several files deleting after confirmation in the dialog box. I did not wait for the process to end, I pressed "delete" again and as a result ALL files in the media folder were deleted.

**Comment by LukeTowers** [comment]
Haven't replicated it for testing just yet but I imagine this would be because the reference passed for what resource should be deleted is somehow being set to the containing folder after the first request.

**Comment by daftspunk** [comment]
Unable to replicate, I receive the error:

`
dev.ERROR: exception 'League\Flysystem\FileNotFoundException' with message 'File not found at path: media/around-glenrowan-150x150.jpg' in ~\vendor\league\flysystem\src\Filesystem.php:385
`

**Comment by plyusninva** [comment]
Try checking on a slow connection.

**Comment by daftspunk** [comment]
That's what I did, I opened Chrome tools and artificially throttled my connection to the speed of a 56k modem. Can anyone else replicate this?

**Comment by plyusninva** [comment]
media/test/test2/
file1.png
file2.png
file3.png

Step 1: select file2
Step 2: click Delete
Step 3: click Ok
Step 4: Deletion process started
Step 5: Not waiting for the completion of the removal process clicked Delete
Step 6: click Ok
Result - folders "test" and "test2" deleted together with files "file1, file2, file3"

![image](https://cloud.githubusercontent.com/assets/25381948/24743501/e572b37e-1ac7-11e7-9c29-8c6f8f416510.png)

**Comment by LukeTowers** [comment]
You'd have to check it on a slow server, the connection only affects the speed of the communication to the server, not how long it takes the server to "boot up" the initial deletion request and then actually delete the files. Try telling PHP to `sleep()` for long enough to send a second request before actually deleting the files.

**Comment by plyusninva** [comment]
need disable button "Delete" when deletion process started

**Comment by daftspunk** [comment]
@LukeTowers were you able to replicate this? Seems very odd that the code would trash the entire contents.

**Comment by LukeTowers** [comment]
@daftspunk Not yet, I'll bump this up on my priority list to look at

**Comment by LukeTowers** [comment]
I was able to replicate this:
Folder Structure:
```
storage/app/media:
- rootFileTest.png
- test
    - file1.png
    - file2.png
    - file3.png
    - test2
        - file1.png
        - file2.png
        - file3.png
```

Selected files:
![screenshot 2017-05-14 10 20 13](https://cloud.githubusercontent.com/assets/7253840/26035774/fc96873a-388e-11e7-91fd-510c7c51b4f3.png)



First delete request posted FormData:
```
search:
sorting:title
paths[0][path]:/test/test2/file1.png
paths[0][type]:file
paths[1][path]:/test/test2/file2.png
paths[1][type]:file
paths[2][path]:/test/test2/file3.png
paths[2][type]:file
```

Second delete request posted FormData: (made immediately after the first request without waiting for it to finish)
```
search:
sorting:title
paths[0][path]:/test
paths[0][type]:folder
```

Result:
Folder Structure:
```
storage/app/media:
- rootFileTest.png
```

**Comment by LukeTowers** [comment]
This problem is caused by two reasons:

1. Upon deleting all the items in a subfolder, the "back to parent folder" (`..`) item is selected 
2. Triggering a delete with this item selected actually works and deletes the parent folder (and its contents).

I don't think that 2 is an expected result and should be what I'm focusing on to prevent.

**Comment by LukeTowers** [comment]
Fixed by #2867
